### ISSUE TITLE
Using existing boot volume instead of creating new one

### ISSUE THREAD

**Issue by nickk289** [mohankumarpaluru/oracle-freetier-instance-creation#1]
Hi, how can I define an already created boot volume (ocid1.bootvolume.oc1.eu-frankfurt-1.ab...) instead of creating a new boot volume by selecting OCI_IMAGE_ID ?

**Comment by mohankumarpaluru** [comment]
The script doesn't support using an existing boot volume, atleast as of now. OCI_IMAGE_ID is for the OS we want to use

**Comment by Nirzak** [comment]
> The script doesn't support using an existing boot volume, atleast as of now. OCI_IMAGE_ID is for the OS we want to use

What will happen if we mention the existing boot volume's IMAGE ID in OCI_IMAGE_ID ? wont't it then use the existing boot volume instead of creating one? thanks

**Comment by nickk289** [comment]
I examined the oci api documentation and tried different variations of code modifications in the **main.py** file where parameters are sent to create an instance until it passed without error. After editing the code below (I commented out the **image_id** variable and added a **source_details** section) I no longer see the error in the console, but I don't know if it really works because I'm still getting an "out of capacity" response from the api.


```
                    display_name=DISPLAY_NAME,
                    shape="VM.Standard.A1.Flex",
#                    image_id=oci_image_id,
                    availability_config=oci.core.models.LaunchInstanceAvailabilityConfigDetails(
                        recovery_action="RESTORE_INSTANCE"
                    ),
                    instance_options=oci.core.models.InstanceOptions(
                        are_legacy_imds_endpoints_disabled=False
                    ),
                    source_details=oci.core.models.InstanceSourceViaBootVolumeDetails(
                        source_type="bootVolume",
                        boot_volume_id="ocid1.bootvolume.oc1.eu-frankfurt-1.ab...."
                    ),
                    shape_config=oci.core.models.LaunchInstanceShapeConfigDetails(
                        ocpus=4, memory_in_gbs=24
                    ),
```

**Comment by mohankumarpaluru** [comment]
Without altering the script, it is currently not feasible to use the image ID of the boot volume.

@nickk289's comment addresses the requirement for both you and him, you can utilize the same code change.

The primary objective of my script is to remain as simple as possible. I do not want to deviate from this goal while attempting to enhance its functionality with additional features and customizations.

Considering the complexity involved in accommodating both Image ID and boot volume image ID, I plan to integrate this into the script,  as the [imageid](https://docs.oracle.com/en-us/iaas/api/#/en/iaas/20160918/Instance/:~:text=of%20the%20instance.-,imageId,-Required%3A%20no) I'm using in [LaunchInstance](https://docs.oracle.com/en-us/iaas/api/#/en/iaas/20160918/datatypes/LaunchInstanceDetails) API appears to be deprecated now, and the use of [sourceDetails](https://docs.oracle.com/en-us/iaas/api/#/en/iaas/20160918/datatypes/InstanceSourceDetails) offers a viable alternative.

I'll keep this open and update it once I decide so others can see it

**Comment by Nirzak** [comment]
Ok got it. even if we able to define the current boot volume to use during instance creation it will not allow us to do as our previous instance was not on VM.Standard.A1.Flex . boot volume's cpu must be matched during using an existing boot volume during instance creation.

**Comment by mohankumarpaluru** [comment]
> boot volume's cpu must be matched if you use the existing boot volume during instance creation.

That's correct. This option to define boot volume image id will be useful only in scenario where one wants to repurpose a boot volume from a deleted ARM instance.

**Comment by deffuseyou** [comment]
> > boot volume's cpu must be matched if you use the existing boot volume during instance creation.
> 
> That's correct. This option to define boot volume image id will be useful only in scenario where one wants to repurpose a boot volume from a deleted ARM instance.

Could you please tell me how the memory is allocated then? I'm using 159 GB for my instance to create an ARM instance? how much is reserved during the execution of your script and can i then expand to the maximum possible size?
